src/machine/machine_atsamd21.go

```diff
@@ -193,40 +193,40 @@ func (a ADC) Get() uint16 {
 	ch := a.getADCChannel()
 
 	// Selection for the positive ADC input
-	sam.ADC.INPUTCTRL &^= sam.ADC_INPUTCTRL_MUXPOS_Msk
+	sam.ADC.INPUTCTRL.ClearBits(sam.ADC_INPUTCTRL_MUXPOS_Msk)
 	waitADCSync()
-	sam.ADC.INPUTCTRL |= sam.RegValue(ch << sam.ADC_INPUTCTRL_MUXPOS_Pos)
+	sam.ADC.INPUTCTRL.SetBits(uint32(ch << sam.ADC_INPUTCTRL_MUXPOS_Pos))
 	waitADCSync()
 
 	// Select internal ground for ADC input
-	sam.ADC.INPUTCTRL &^= sam.ADC_INPUTCTRL_MUXNEG_Msk
+	sam.ADC.INPUTCTRL.ClearBits(sam.ADC_INPUTCTRL_MUXNEG_Msk)
 	waitADCSync()
-	sam.ADC.INPUTCTRL |= sam.RegValue(sam.ADC_INPUTCTRL_MUXNEG_GND << sam.ADC_INPUTCTRL_MUXNEG_Pos)
+	sam.ADC.INPUTCTRL.SetBits(sam.ADC_INPUTCTRL_MUXNEG_GND << sam.ADC_INPUTCTRL_MUXNEG_Pos)
 	waitADCSync()
 
 	// Enable ADC
-	sam.ADC.CTRLA |= sam.ADC_CTRLA_ENABLE
+	sam.ADC.CTRLA.SetBits(sam.ADC_CTRLA_ENABLE)
 	waitADCSync()
 
 	// Start conversion
-	sam.ADC.SWTRIG |= sam.ADC_SWTRIG_START
+	sam.ADC.SWTRIG.SetBits(sam.ADC_SWTRIG_START)
 	waitADCSync()
 
 	// Clear the Data Ready flag
-	sam.ADC.INTFLAG = sam.ADC_INTFLAG_RESRDY
+	sam.ADC.INTFLAG.SetBits(sam.ADC_INTFLAG_RESRDY)
 	waitADCSync()
 
 	// Start conversion again, since first conversion after reference voltage changed is invalid.
-	sam.ADC.SWTRIG |= sam.ADC_SWTRIG_START
+	sam.ADC.SWTRIG.SetBits(sam.ADC_SWTRIG_START)
 	waitADCSync()
 
 	// Waiting for conversion to complete
-	for (sam.ADC.INTFLAG & sam.ADC_INTFLAG_RESRDY) == 0 {
+	for (sam.ADC.INTFLAG.Get() & sam.ADC_INTFLAG_RESRDY) == 0 {
```

Comment:
It might even be useful to have a `.HasBits(sam.ADC_INTFLAG_RESRDY)` that just returns a boolean whether all the bits are set. Not terribly important though.

Reply:
That might be a good idea. However in this case we would need `.WithoutBits()` since we are testing for it *not* being set.

Reply:
Heh, of course.

```go
	for (!sam.ADC.INTFLAG.HasBits(sam.ADC_INTFLAG_RESRDY) {
 ```

Reply:
So, should we do this? If so, now is the time.

Reply:
I have no strong preference either way, it's basically just ease of use. So I leave it up to you if you want to implement it or not.

Reply:
I will address it in a future iteration then, since it will not break anything, just reduce code.